test/core/linter_test.py

```diff
@@ -708,18 +708,42 @@ def test_linter_noqa_with_templating():
     assert not result.get_violations()
 
 
+def test_linter_noqa_template_errors():
+    """Similar to test_linter_noqa, but uses templating (Jinja)."""
+    lntr = Linter(
+        config=FluffConfig(
+            overrides={
+                "templater": "jinja",
+            }
+        )
+    )
+    sql = """select * --noqa: TMP
```

Comment:
why do we need this "noqa" comment here?

Reply:
When templating fails due to an undefined variable, Jinja throws an exception but doesn't tell us what line it happened on. In this PR, I updated it so a templating error is reported with a line number of 1. Since we don't know where the error occurred, this is better than nothing. A user will see the error reported at line 1, then they'll know they need to add a "noqa" comment at line 1. This whole workflow feels a little weird -- as you pointed out in one of the original issues, they could (and probably should) just fix their templating variables. But if somebody wants to ignore these errors, now they can.

Reply:
OK so you need it on BOTH line 1 AND the actual failure line by the looks of things.

This looks like something we should mention in https://docs.sqlfluff.com/en/stable/rules.html#inline-ignoring-errors and

Reply:
In some cases, yes. It depends on the error. In this case, it was something being "undefined". There are other kinds of errors (e.g. the template code raises an exception or has a syntax error, e.g. missing parenthesis. I'm hesitant to mention it in the docs, because it's situation dependent. If someone needs to do it, they'll know they need to do it because they'll run `sqlfluff lint` and see the message refers to line 1. This seems self explanatory to me. Have you (can you) try linting the above example using this branch, but first remove all the `--noqa` comments? I think you'll see what I mean. If you still feel like the doc change is needed, I'm happy to add it then.

Reply:
Yeah that's what I ran to get my mind sorted.

OK if it's not consistent then happy to leave for now. I agree `lint` does point you in that direction anyway.

Resolving.